app/src/main/java/ch/epfl/sdp/game/GameThread.java

```diff
@@ -0,0 +1,81 @@
+package ch.epfl.sdp.game;
+
+/**
+ * Manages the main loop
+ */
+public class GameThread extends Thread{
+    private int FPS = 30;
+    private double avgFPS;
+    private boolean running;
+    private Game game;
+
+    public GameThread(Game game){
+        this.game = game;
+    }
+
+    /**
+     * Get the actual state of the loop
+     * @return true if it is running, false otherwise
+     */
+    public boolean isRunning() {
+        return running;
+    }
+
+    /**
+     * Set true to start or continue the loop, false to stop it.
+     * @param running defines the state of the loop
+     */
+    public void setRunning(boolean running) {
+        this.running = running;
+    }
+
+    /**
+     * Main loop that can be parameterized, started and stoped, manages frames and updates
+     */
+    @Override
+    public void run(){
+        long startTime;
+        long timeMillis;
+        long waitTime;
+        long totalTime = 0;
+        int frameCount = 0;
+        long targetTime = 1000 / FPS;
+
+        // Main loop of the game
+        while(running){
+            startTime = System.nanoTime();
+
+            // Does this needs synchronization?
```

Comment:
I'm not totally sure. The draw method should not take too much time if we correctly use the canvas